pkg/pool-utils/contracts/external-fees/ProtocolFeeCache.sol

```diff
@@ -131,18 +136,23 @@ abstract contract ProtocolFeeCache is RecoveryMode {
         // solhint-disable-previous-line no-empty-blocks
     }
 
-    function _updateProtocolFeeCache(IProtocolFeePercentagesProvider protocolFeeProvider, ProviderFeeIDs memory feeIds)
-        private
-    {
-        uint256 swapFee = protocolFeeProvider.getFeeTypePercentage(feeIds.swap);
-        uint256 yieldFee = protocolFeeProvider.getFeeTypePercentage(feeIds.yield);
-        uint256 aumFee = protocolFeeProvider.getFeeTypePercentage(feeIds.aum);
-
-        _cache = FeeTypeCache({
-            swapFee: swapFee.toUint64(),
-            yieldFee: yieldFee.toUint64(),
-            aumFee: aumFee.toUint64()
-        });
-        emit ProtocolFeePercentageCacheUpdated(_cache);
+    function _updateProtocolFeeCache(IProtocolFeePercentagesProvider protocolFeeProvider, bytes32 feeIds) private {
+        uint256 swapFee = protocolFeeProvider.getFeeTypePercentage(
+            feeIds.decodeUint(_SWAP_FEE_ID_OFFSET, _FEE_TYPE_ID_WIDTH)
```

Comment:
```suggestion
            getProviderFeeId(ProtocolFeeType.SWAP)
```

This likely saves bytecode as well

Reply:
This doesn't work because that would read an immutable member during construction.
Adding a version of `getProviderFeeId` that takes the IDs as parameter doesn't save anything.